src/coreclr/jit/block.h

```diff
@@ -606,98 +616,164 @@ struct BasicBlock : private LIR::Range
         return KindIs(BBJ_ALWAYS, BBJ_CALLFINALLY, BBJ_COND, BBJ_EHCATCHRET, BBJ_EHFILTERRET, BBJ_LEAVE);
     }
 
-    BasicBlock* GetJumpDest() const
+    BasicBlock* GetTarget() const
+    {
+        // BBJ_COND should use GetTrueTarget
+        assert(!KindIs(BBJ_COND));
+
+        // If bbKind indicates this block has a jump, bbTarget cannot be null
+        assert(!HasTarget() || HasInitializedTarget());
+        return bbTarget;
+    }
+
+    void SetTarget(BasicBlock* target)
+    {
+        // BBJ_COND should use SetTrueTarget
+        assert(!KindIs(BBJ_COND));
+
+        // SetKindAndTarget() nulls target for non-jump kinds,
+        // so don't use SetTarget() to null bbTarget without updating bbKind.
+        bbTarget = target;
+        assert(!HasTarget() || HasInitializedTarget());
+    }
+
+    BasicBlock* GetTrueTarget() const
     {
-        // If bbJumpKind indicates this block has a jump, bbJumpDest cannot be null
-        assert(!HasJumpDest() || HasInitializedJumpDest());
-        return bbJumpDest;
+        assert(KindIs(BBJ_COND));
+        assert(HasInitializedTarget());
+        return bbTrueTarget;
     }
 
-    void SetJumpDest(BasicBlock* jumpDest)
+    void SetTrueTarget(BasicBlock* target)
     {
-        // SetJumpKindAndTarget() nulls jumpDest for non-jump kinds,
-        // so don't use SetJumpDest() to null bbJumpDest without updating bbJumpKind.
-        bbJumpDest = jumpDest;
-        assert(!HasJumpDest() || HasInitializedJumpDest());
+        assert(KindIs(BBJ_COND));
+        assert(target != nullptr);
+        bbTrueTarget = target;
     }
 
-    void SetJumpKindAndTarget(BBjumpKinds jumpKind, BasicBlock* jumpDest = nullptr)
+    bool TrueTargetIs(const BasicBlock* target) const
     {
-        bbJumpKind = jumpKind;
-        bbJumpDest = jumpDest;
+        assert(KindIs(BBJ_COND));
+        assert(HasInitializedTarget());
+        assert(target != nullptr);
+        return (bbTrueTarget == target);
+    }
+
+    BasicBlock* GetFalseTarget() const
+    {
+        assert(KindIs(BBJ_COND));
+
+        // So long as bbFalseTarget tracks bbNext in SetNext(), it is possible for bbFalseTarget to be null
+        // if this block is unlinked from the block list.
+        // So check bbNext before triggering the assert if bbFalseTarget is null.
+        // TODO-NoFallThrough: Remove IsLast() check once bbFalseTarget isn't hard-coded to bbNext
+        assert((bbFalseTarget != nullptr) || IsLast());
+        return bbFalseTarget;
+    }
+
+    void SetFalseTarget(BasicBlock* target)
+    {
+        assert(KindIs(BBJ_COND));
+        assert(target != nullptr);
+        bbFalseTarget = target;
+    }
+
+    bool FalseTargetIs(const BasicBlock* target) const
+    {
+        assert(KindIs(BBJ_COND));
+        assert(bbFalseTarget != nullptr);
+        assert(target != nullptr);
+        return (bbFalseTarget == target);
+    }
+
+    void SetCondKindAndTarget(BasicBlock* target)
```

Comment:
This doesn't exactly roll off the tongue -- why not `SetCond`? Eventually we would hopefully end up with just `SetCond(BasicBlock* falseTarget, BasicBlock* trueTarget)`, `SetSwitch(BBswtDesc* swtTarget)` and so on.

Reply:
I can shorten the names of all the `SetKind...` variants. Would you be ok with `SetKindAndTarget` being left as-is? I'm not sure how we can intuitively shorten that one while still conveying it's setting the jump kind and target.

Reply:
Yes, I think `SetKindAndTarget` makes sense -- it's sort of a low-level operation, while these other ones are more high-level.